src/effects/Effect.ts

```diff
@@ -0,0 +1,4 @@
+export class Effect<T> {
+  constructor(readonly value: T) {}
+}
+export const effect = <T>(value: T) => new Effect(value)
```

Comment:
I'm not making this function public yet - `Effect` is public because with the current state consumers might need `instanceof Effect`